api/handlers/dataexplorerhandler.py

```diff
@@ -251,6 +251,91 @@ def _parse_request(self, request_type='search'):
 
         return return_type, modified_filters, search_string
 
+    @require_login
+    def get_type_ahead_values(self):
+        """
+        Return list of type ahead values for a key given a value 
+        that the user has already started to type in for the value of 
+        a custom string field
+        """
+        user_value = self.request.json_body['value']
+        custom_field = self.request.json_body['field_name']
+        modified_filters = []
+        # Add permissions filter to list if user is not requesting all data
+        if not self.request.json_body.get('all_data', False):
+            modified_filters.append({'term': {'permissions._id': self.uid}})
+
+        body = {
+            "size": 0, 
+            "query": {
+                "bool": {
+                    "must" : {
+                        "match" : { custom_field : user_value}
+                    },
+                    "filter" : {
+                        "bool" : {
+                            "must" : modified_filters
+                        }
+                    }
+                    
+                }
+            },
+            "aggs" : {
+                "results" : {
+                    "terms" : {
+                        "field" : custom_field + ".raw",
+                        "size" : 15
+                    }
+                }
+            }
+        }
+
+        aggs = config.es.search(
+            index='data_explorer',
+            doc_type='flywheel',
+            body=body
+        )['aggregations']['results']['buckets']
+        aggs = [bucket['key'] for bucket in aggs]
+        return {'type_aheads': aggs}
+
+    def get_type_ahead_numbers(self):
+        # user_value = self.request.json_body['value']
```

Comment:
If we add a lookup to the data_explorer_fields index to discover the type, we could just have one function and one endpoint, `aggregate`. It's also not really a type-ahead for numbers - it's more of a stats endpoint for a specific field, the user would not be providing a search string. 

I think a good interface for this endpoint would be:
```
{
    'search_string': '<the search string provided by the user, used for type ahead on string values>', #Not required
    'field_name': '<full name of field>"
}
```

And then a lookup to see what type the field name is, using a stats agg if `integer`, `float`, or `date`, and a terms agg if `string`. If `string`, and no `search_string`, remove the filter on the `search_string` from the query.